### ISSUE TITLE
How to generate labels to fine-tune SuperPoint

### ISSUE THREAD

**Issue by zzkslam** [rpautrat/SuperPoint#204]
Hi,thanks for your great work.If I want to fine-tune SuperPoint with my own dataset,how to generate labels as **superpoint_coco.yaml**requires?
![2021-04-24 23-02-19屏幕截图](https://user-images.githubusercontent.com/57522526/115963437-da663f00-a551-11eb-92f2-d8407ba04856.png)

**Comment by rpautrat** [comment]
Hi,

You would need to follow the step 2 of the Readme, but adapt it to your own dataset instead of COCO. You basically need to add a dataloader for your own dataset, by following the structure of `datasets/coco.py`. You can use the pretrained model provided in the Readme to export the detections.

**Comment by zzkslam** [comment]
OK,I got it.Thank you very much

**Comment by zzkslam** [comment]
I notice in _coco.py_,images are resized to 240*360 by default.You also noted that MagicPoint and SuperPoint only work on images with dimensions divisible by 8 and the user is responsible for resizing them to a valid dimension.If I only change the base_path in _coco.py_to  the path of my own dataset(image size:376*1241),then I don't need to resize the image since the image will be rezised to 240*360 by default.Is it correct and anything else should I change to retrain in my own dataset?

**Comment by rpautrat** [comment]
Hi, this is indeed correct. Your images will be automatically resized to 240x360 (by cropping the original image to get the same aspect ratio, and then resizing it to the right size). As long as your data structure is the same as in COCO (all images in the same base folder), then yes, I think that all you need to change is the base path to your own dataset.

**Comment by zzkslam** [comment]
OK,thank you very much

**Comment by zzkslam** [comment]
Sorry to bother you again.I also want to know if I train magic point from scratch, then in the step 3, when I run p**ython experiment.py train configs/magic-point_coco_train.yaml magic-point_coco**, Do I need to add **--pretrained model magic-point_synth**(magic-point_synth is the model generated on Synthetic Shapes in step 1).Have you tried this way to train magic point on MS-COCO?

**Comment by rpautrat** [comment]
I was always training MagicPoint on the COCO dataset from scratch, without reusing the pretrained model on the synthetic dataset. But in theory you can indeed add the `--pretrained model magic-point_synth` option, this should accelerate the training time I think.

**Comment by zzkslam** [comment]
OK,thank you for your advice.

**Comment by zzkslam** [comment]
In the fourth step,When I run **detector_repeatability_hpatches.ipynb** , I met the problem like this:
![Screenshot from 2021-05-05 21-49-46](https://user-images.githubusercontent.com/57522526/117151629-d8bb3780-adeb-11eb-9a48-38ba55d9a971.png)
It can show the pictures and keypoints like this:
![Screenshot from 2021-05-05 21-53-40](https://user-images.githubusercontent.com/57522526/117152300-79a9f280-adec-11eb-9649-bd5daf9925f4.png)

Can you tell me how to solve this problem?And you also said **You might repeat steps 2) and 3) several times**.When I repeat steps 2),should I use **magic-point_synth** or **magic-point_coco** to generate these labels? When I repeat steps 3),should I add --pretrained model magic-point_coco?

**Comment by rpautrat** [comment]
The warning indicates a mean of an empty list, which should be the cause of the nan values. The images and detections look good though. Something must go wrong somewhere and the repeatability list is somehow never updated. My guess is that the path or name of your experiment is wrong and in the [compute_repeatability](https://github.com/rpautrat/SuperPoint/blob/313a0a115961bc2a98a5ed1d7bde33fa99542398/superpoint/evaluations/detector_evaluation.py#L137) function, retrieving the paths gives you an empty list: https://github.com/rpautrat/SuperPoint/blob/313a0a115961bc2a98a5ed1d7bde33fa99542398/superpoint/evaluations/detector_evaluation.py#L174.

The first time that you run step 2), you should use **magic-point_synth** (because you only have this one available at that point). However, the second time you run step 2), you should use your lastly trained model, i.e. **magic-point_coco**.
You can indeed add `--pretrained model magic-point_synth` in the first round and `--pretrained model magic-point_coco` in the second round. This will allow the trainings to converge faster.

**Comment by zzkslam** [comment]
OK,I got it.Indeed this is something wrong with my path.Thanks for your great help!

**Comment by zzkslam** [comment]
When I retrained the superpoint, I found its performance is not good facing illumination change and I also noticed that in **superpoint_coco.yaml**, there is some parameters I can change to improve the robustness of superpoint's descriptors.Can you teach me how to use these parameters to improve the robustness of superpoint's descriptors, or say to what extent should I modift these parameters?
![Screenshot from 2021-05-07 19-26-23](https://user-images.githubusercontent.com/57522526/117443684-16929a00-af6b-11eb-80ca-3fe6b4b5c2d6.png)

**Comment by rpautrat** [comment]
I don't think you will get much improvement by tuning these parameters. You can try to make the photometric augmentation harder by tuning the parameters and visualizing the changes in this notebook: https://github.com/rpautrat/SuperPoint/blob/master/notebooks/visualize_coco_augmentation.ipynb.

But to get a really better illumination invariance, you should get pairs of real images with different lighting as explained [here](https://github.com/rpautrat/SuperPoint/issues/188#issuecomment-765200969). But I think that you already tried that, right?

**Comment by zzkslam** [comment]
Yes, I used the first link of dataset(A Dataset of Multi-Illumination Images in the Wild) you provided to retrain superpoint. I just gather these pictures to a folder  and change the base path in **coco.py** to this folder and then run:**python superpoint/experiment.py train superpoint/configs/superpoint_coco.yaml superpoint_finetuned --pretrained_model sp_v6** using the labels generated by step 2.I also noticed that in **superpoint_coco.yaml**,homography adaptation has not been completed,do you mean I need to implement homography adaptation in **superpoint_coco.yaml** according to **magic-point_coco_train.yaml**?

**Comment by rpautrat** [comment]
No, you don't need homography adaptation to train SuperPoint. This is only useful to export the ground truth in step 2.

So I don't think there is much more that you can do to improve robustness to illumination. This will anyway never be perfect.

But why results do you actually get currently? Are they really bad, or just similar to the original SuperPoint?

**Comment by zzkslam** [comment]
I want to insert superpoint in SLAM structure to extract keypoints and descriptors.I retrianed superpoint on KITTI dataset(model one) and get a little improvement, however,when illumation change in a some extent,for example, running through shadows,it is likely to lose track. So I want to improve the descriptors' robustness to illumination.Then I used first link of dataset(A Dataset of Multi-Illumination Images in the Wild) you provided to retrain superpoint(using model one as pretrained model) for 30000 iterations.Then the performance decreased to a little bit.I thought it may be caused by the different contents between KITTI and the dataset you provided

**Comment by rpautrat** [comment]
Adding more illumination invariance also makes the descriptor less robust overall, because it becomes too general. There is no perfect way to handle illumination changes correctly anyway, this is an open research area.

**Comment by zzkslam** [comment]
OK,Thank you again for your useful advice and great patience.
